optaplanner-core/src/test/java/org/optaplanner/core/impl/testdata/domain/comparable/TestdataCodeComparator.java

```diff
@@ -0,0 +1,15 @@
+package org.optaplanner.core.impl.testdata.domain.comparable;
+
+import java.util.Comparator;
+
+import org.apache.commons.lang3.builder.CompareToBuilder;
+import org.optaplanner.core.impl.testdata.domain.TestdataObject;
+
+public class TestdataCodeComparator implements Comparator<TestdataObject> {
+
+    public int compare(TestdataObject a, TestdataObject b) {
+        return new CompareToBuilder()
+                .append(a.getCode(), b.getCode())
+                .toComparison();
```

Comment:
Please Comparator.comparing().thenComparing() instead and avoid having to create the class to begin with.

Reply:
I've created the comparator in order to have a difficulty comparator class which I can put into the @PlanningEntity(difficultyComparatorClass = TestdataCodeComparator.class) annotation above the comparable entity. Having a comparable entities by difficulty is mandatory when using EntitySorterManner.DECREASING_DIFFICULTY in the ExhaustiveSearchPhaseConfig, that's why it's there. Is there a simpler way how to achieve this without needing to create a Comparator class that you explicitly put into the annotation?

Reply:
@ge0ffrey I agree with @michaltomco regarding the necessity of this `Comparator` class. The usage of the `CompareToBuilder` can be replaced by standard `java.util.Comparator`, though. See e.g. https://github.com/kiegroup/optaplanner/blob/master/optaplanner-examples/src/main/java/org/optaplanner/examples/cloudbalancing/optional/domain/CloudProcessDifficultyComparator.java#L24